digest.go

```diff
@@ -109,18 +109,11 @@ func (d Digest) Validate() error {
 		return ErrDigestInvalidFormat
```

Comment:
This regexp check is no longer needed, the second regexp will do all the needed validation and the algorithm lookup validates the algorithm.